dms/models/directory.py

```diff
@@ -469,12 +459,19 @@ def _compute_groups(self):
     # ----------------------------------------------------------
 
     @api.depends("is_root_directory")
-    def _compute_directory_type(self):
+    def _compute_parent_id(self):
         for record in self:
             if record.is_root_directory:
                 record.parent_id = None
             else:
-                record.root_storage_id = None
+                if self.env.context.get(
```

Comment:
This seems more a default value than something that should be put in a computed

Reply:
I have changed now to default and onchange and i have removed compute.

Reply:
Well, the compute can be correct for avoiding the onchange. My comment was about the default value. I think you can keep the compute for the other part. Or is there any problem on that?

Reply:
Yes, if we keep compute function exists some possibilities: 
- Option A: Only compute values according to `is_root_directory=True`. In these case appear error describe in these PR.
- Option B: Compute values according to `is_root_directory=True` and in other cases (else) call to default function.
- Option C: Remove compute but NOT create onchange function. If we created directories (specially in tests) show constraint according to is set `parent_id` in root directories.

Do you think in other option?